### ISSUE TITLE
 TypeError: run() got an unexpected keyword argument 'mem_reservaction'

### ISSUE THREAD

**Issue by godfriedmeesters** [CybercentreCanada/assemblyline-core#51]
After adding my new service (named 'Whitelist') and submitting a file, scaler gives the following error:

```
2021-01-22 11:51:11,747  WARNING 2afd534beacd 1            assemblyline.scaler | Restarting thread: Service Configuration Sync
2021-01-22 11:51:11,823    ERROR 2afd534beacd 1            assemblyline.scaler | Error applying service settings from: Whitelist

Traceback (most recent call last):
  File "/root/.local/lib/python3.7/site-packages/assemblyline_core/scaler/scaler_server.py", line 340, in sync_services
    labels={'dependency_for': service.name}
  File "/root/.local/lib/python3.7/site-packages/assemblyline_core/scaler/controllers/docker_ctl.py", line 330, in start_stateful_container
    cfg=spec.container, network=self._get_network(service_name).name)
  File "/root/.local/lib/python3.7/site-packages/assemblyline_core/scaler/controllers/docker_ctl.py", line 192, in _start_container
    ports=ports,
  File "/root/.local/lib/python3.7/site-packages/docker/models/containers.py", line 812, in run
    detach=detach, **kwargs)
  File "/root/.local/lib/python3.7/site-packages/docker/models/containers.py", line 869, in create
    create_kwargs = _create_container_args(kwargs)
  File "/root/.local/lib/python3.7/site-packages/docker/models/containers.py", line 1085, in _create_container_args
    raise create_unexpected_kwargs_error('run', kwargs)
TypeError: run() got an unexpected keyword argument 'mem_reservaction'
2021-01-22 11:51:11,823    ERROR 2afd534beacd 1            assemblyline.scaler | Crash in scaler: sync_services

```

**Comment by godfriedmeesters** [comment]
My service_manifest.yml:

```
name: Whitelist
version: 1
description: >
  ALv4 Whitelist service

  This service will calculate a SHA1 hash of a submitted file, then lookup this hash in a local Redis set of known good hashes.
  The service will check https://www.nist.gov/itl/ssd/software-quality-group/national-software-reference-library-nsrl/nsrl-download/current-rds
  for new hashes automatically every 24 hrs.

accepts: .*
rejects: empty|metadata/.*

stage: FILTER
category: Filtering

file_required: true
# Maximum execution time the service has before it's considered to be timed out
timeout: 10
disable_cache: false

# is the service enabled by default
enabled: true

docker_config:
  image: godfried/whitelist-service:latest
  cpu_cores: 1.0
  ram_mb: 1024

dependencies:
  redisdb:
    container:
      image: redis

update_config:
  update_interval_seconds: 3600
  generates_signatures: false
  method: run
  run_options:
    allow_internet_access: true
    command: ["python",  "whitelist_updater.py"]
    image: godfried/whitelist-service:latest 
```

**Comment by cccs-douglass** [comment]
Fixed in PRs #57 #56

**Comment by godfriedmeesters** [comment]
Thanks

Op do 28 jan. 2021 16:29 schreef Adam <notifications@github.com>:

**Comment by cccs-douglass** [comment]
I'm still getting used to our new branching model, I goofed those ones, its #58 #59 now

**Comment by cccs-douglass** [comment]
Fixed in the version 4.0.0.16 build.
